pkg/operators/kubenameresolver/kubenameresolver.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package kubenameresolver provides an operator that enriches events by looking
+// up the pod name and namespace and enriches it with its ip information. It is
+// currently used by the following gadgets:
+// - trace network
+package kubenameresolver
+
+import (
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/common"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/kubemanager"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	OperatorName = "KubeNameResolver"
+)
+
+type KubeNameResolverInterface interface {
+	SetLocalPodDetails(owner, hostIP, podIP string, labels map[string]string)
+}
+
+type KubeNameResolver struct {
+	k8sInventory *common.K8sInventoryCache
+}
+
+func (k *KubeNameResolver) Name() string {
+	return OperatorName
+}
+
+func (k *KubeNameResolver) Description() string {
+	return "KubeNameResolver resolves pod name/namespace to IP addresses..."
+}
+
+func (k *KubeNameResolver) GlobalParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (k *KubeNameResolver) ParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (k *KubeNameResolver) Dependencies() []string {
+	return []string{kubemanager.OperatorName}
+}
+
+func (k *KubeNameResolver) CanOperateOn(gadget gadgets.GadgetDesc) bool {
+	km := kubemanager.KubeManager{}
+	if !km.CanOperateOn(gadget) {
+		return false
+	}
+	_, hasNameResolverInterface := gadget.EventPrototype().(KubeNameResolverInterface)
+	return hasNameResolverInterface
+}
+
+func (k *KubeNameResolver) Init(params *params.Params) error {
+	log.Warnf("Init %s", OperatorName)
+	k8sInventory := common.GetK8sInventoryCache()
+	k.k8sInventory = k8sInventory
+	return nil
+}
+
+func (k *KubeNameResolver) Close() error {
+	k.k8sInventory.Close()
+	return nil
+}
+
+func (k *KubeNameResolver) Instantiate(gadgetCtx operators.GadgetContext, gadgetInstance any, params *params.Params) (operators.OperatorInstance, error) {
+	log.Warnf("Instantiate %s", OperatorName)
```

Comment:
dito

In particular, for the instance you have the gadget context, you could do the following:

```suggestion
	gadgetCtx.Logger().Debugf("Instantiating %s", OperatorName)
```

BTW, We need a way to pass the logger to the operator as well. cc @flyth

Reply:
Everything not running in the instance should use a server side logger - I think for now we could use `logger.DefaultLogger()`, but it would be great if we could somehow pass around an already initialized logger, maybe even with prefixes / additional fields (for the operator using it in this case).

/cc @mqasimsarfraz as he wanted to improve the logger as well :)